packages/http/src/mocker/generator/HttpParamGenerator.ts

```diff
@@ -15,27 +15,15 @@ function improveSchema(schema: JSONSchema) {
     if (!newSchema.maximum) {
       newSchema.maximum = 1000;
     }
-  }
-
-  if (newSchema.type === 'string') {
-    if (!newSchema.format && !newSchema.enum) {
-      newSchema['x-faker'] = 'lorem.word';
-    }
-  }
-
-  if (newSchema.type === 'object') {
-    if (newSchema.properties) {
-      newSchema.properties = Object.entries(newSchema.properties).reduce((r, [k, v]) => {
-        r[k] = improveSchema(v);
-        return r;
-      }, {});
-    }
-  }
-
-  if (newSchema.type === 'array') {
-    if (typeof newSchema.items === 'object') {
-      newSchema.items = improveSchema(newSchema.items);
-    }
+  } else if (newSchema.type === 'string' && !newSchema.format && !newSchema.enum) {
+    newSchema['x-faker'] = 'lorem.word';
+  } else if (newSchema.type === 'object' && newSchema.properties) {
+    newSchema.properties = Object.entries(newSchema.properties).reduce((r, [k, v]) => {
+      r[k] = improveSchema(v);
+      return r;
+    }, {});
+  } else if (newSchema.type === 'array' && typeof newSchema.items === 'object') {
+    newSchema.items = improveSchema(newSchema.items);
```

Comment:
It sure does look complicated. Can we add some tests to it? I've checked the coverage for this and it might need some additional tests. Coverage is lacking for the "red lines":
![coverage](https://user-images.githubusercontent.com/24682079/66496962-0cbd6200-eabc-11e9-8fea-36dd3c7f8d7a.png)

Reply:
Yeah I agree with you that this has been a little bit under-tested. I've added some more tests for this. 923bee5